src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/IOpenIdConnectEvents.cs

```diff
@@ -38,17 +28,22 @@ public interface IOpenIdConnectEvents : IRemoteAuthenticationEvents
         /// <summary>
         /// Invoked before redirecting to the identity provider to authenticate.
         /// </summary>
-        Task RedirectToAuthenticationEndpoint(RedirectContext context);
+        Task RedirectToIdentityProvider(RedirectContext context);
```

Comment:
Like @HaoK, I really think the events model is for advanced scenarios/users (TBH, it's not the easiest thing to use, though it makes total sense when you understand how it works).

Since it's advanced stuff, I think we can safely assume that the developers using them are familiar enough with OIDC to directly alter the authorization message (if you don't know anything about OIDC, playing with `OpenIdConnectMessage` seems a bit risky anyway).

In this case, I don't see why this event should have a super user-friendly name, since it's not meant for average users.

If you have particular scenarios that require using this event in mind, maybe you should list them here to determine if we can't introduce new first-class properties. Personally, I'm not opposed to having more options, even if they are mainly AAD concepts (as long as they are general enough to be reused by other providers, e.g the `Resource` property that determines the audience of the access tokens returned by AAD)